core/util.py

```diff
@@ -157,3 +224,29 @@ def custom_serialiser(obj: Any) -> str:
         return obj.isoformat()
     else:
         raise TypeError(f"Cannot serialise object of type {type(obj)}")
+
+
+def move_data_to_json_blob(
```

Comment:
we should probably be careful with naming conventions here. JSONB does not actually stand for JSON Blob (it's actually JSON Binary), but we are using it as "blob storage".

Probably a better name would be jsonb_blob, just to be clear?